spec/models/vm_or_template_spec.rb

```diff
@@ -1,6 +1,87 @@
 require "spec_helper"
 
 describe VmOrTemplate do
+  context ".post_refresh_ems" do
+    before do
+      #          fa
+      #       fb
+      #    fc
+      # vc
+      #
+      @before_time = Time.now.utc
+
+      _guid, _server, zone = EvmSpecHelper.local_guid_miq_server_zone
+      @ems      = FactoryGirl.create(:ems_vmware, :zone => zone)
+      @vm       = FactoryGirl.create(:vm_vmware,  :ext_management_system => @ems)
+      @folder_a = FactoryGirl.create(:ems_folder, :ext_management_system => @ems)
+      @folder_b = FactoryGirl.create(:ems_folder, :ext_management_system => @ems)
+      @folder_c = FactoryGirl.create(:ems_folder, :ext_management_system => @ems)
+      @ems.add_child(@folder_a)
+      @folder_a.add_child(@folder_b)
+      @folder_b.add_child(@folder_c)
+      @folder_c.add_child(@vm)
+
+      # post_refresh_ems checks if updated_on >= a provided start time, so jump
+      # forward to avoid subsecond rounding issues.
+      Timecop.travel(1.second)
+      @after_time = Time.now.utc
+    end
+
+    after do
+      Timecop.return
+    end
+
+    def assert_queued_method_for_vm(vm_id, method_name)
+      expected_attributes = {:class_name  => "VmOrTemplate"}
+      if vm_id.kind_of?(Array)
+        expected_attributes[:args] = vm_id
+      else
+        expected_attributes[:instance_id] = vm_id
+      end
+
+      messages = MiqQueue.where(:method_name => method_name).to_a
+      expect(messages.length).to eq 1
+      expect(messages.first).to have_attributes(expected_attributes)
+    end
+
+    it "queues post_create_actions for a new vm" do
+      VmOrTemplate.post_refresh_ems(@ems.id, @before_time)
+      assert_queued_method_for_vm(@vm.id, :post_create_actions)
+    end
+
+    it "queues assign_ems_created_on for a new vm if capture_vm_created_on_date is enabled " do
+      VMDB::Config.stub(:new).and_return(double(:config => {:ems_refresh => {:capture_vm_created_on_date => true}}))
```

Comment:
Also, is there a negative test for this...that is, if the config is false, we won't expect a queue item?